caput/mpiarray.py

```diff
@@ -1241,121 +1303,122 @@ def to_zarr(
             Name or identifier of HDF5 compression filter.
-        compression_opts
+        compression_opts : tuple | None, optional
             See HDF5 documentation for compression filters.
             Compression options for the dataset.
-        file_format : :class:`fileforats.FileFormat`
-            File format interface class
+        file_format : fileformats.FileFormat
+            Instance of :py:class:`~caput.fileformats.FileFormat` specifying
+            the file format to use. Default :py:class:`~caput.fileformats.HDF5`.
         """
         if chunks is None and hasattr(self, "chunks"):
-            logger.error(f"getting chunking opts from mpiarray: {self.chunks}")
+            logger.debug(f"getting chunking opts from mpiarray: {self.chunks}")
             chunks = self.chunks
-        if compression is None and hasattr(self, "compression"):
-            logger.error(f"getting compression opts from mpiarray: {self.compression}")
 
+        if compression is None and hasattr(self, "compression"):
+            logger.debug(f"getting compression opts from mpiarray: {self.compression}")
             compression = self.compression
+
         if compression_opts is None and hasattr(self, "compression_opts"):
-            logger.error(
+            logger.debug(
                 f"getting compression_opts opts from mpiarray: {self.compression_opts}"
             )
-
             compression_opts = self.compression_opts
+
         if file_format == fileformats.HDF5:
             self.to_hdf5(f, dataset, create, chunks, compression, compression_opts)
         elif file_format == fileformats.Zarr:
             self.to_zarr(f, dataset, create, chunks, compression, compression_opts)
         else:
             raise ValueError(f"Unknown file format: {file_format}")
 
-    def _make_selections(self):
+    def _make_selections(self) -> list[slice | int]:
         """Make selections for writing local data to distributed file."""
-        start = self.local_offset[self.axis]
-        end = start + self.local_shape[self.axis]
-
         # Construct slices for axis
-        sel = ([slice(None, None)] * self.axis) + [slice(start, end)]
+        sel = ([slice(None, None)] * self.axis) + [self.local_bounds]
+
         return _expand_sel(sel, self.ndim)
 
-    def transpose(self, *axes):
-        """Transpose the array axes.
+    def transpose(self, *axes: Sequence[int] | int | None) -> MPIArray:  # noqa: D417
+        r"""Transpose the array axes.
 
         Parameters
         ----------
-        axes : None, tuple of ints, or n ints
+        \*axes : Sequence[int] | int | None, optional
+            Transpose arguments:
+
             - None or no argument: reverses the order of the axes.
-            - tuple of ints: i in the j-th place in the tuple means a`s i-th axis
+            - sequence of ints: i in the j-th place in the tuple means a`s i-th axis
               becomes a.transpose()`s j-th axis.
```

Comment:
```suggestion
            - sequence of ints: `i` in the `j`th place in the tuple means `a`'s `i`th axis
              becomes ``a.transpose()``'s `j`th axis.
```
Not really sure how best to format this.  Maybe more double backticks would be better?